tests-arquillian/src/test/java/org/jboss/weld/tests/event/ObservingBean.java

```diff
@@ -0,0 +1,92 @@
+package org.jboss.weld.tests.event;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.event.ObservesAsync;
+import jakarta.enterprise.inject.Default;
+import jakarta.enterprise.inject.spi.EventMetadata;
+
+@ApplicationScoped
+public class ObservingBean {
+
+    private int defaultObjectNotified = 0;
+    private int defaultObjectAsyncNotified = 0;
```

Comment:
These fields should be `volatile`, at least for the async events where the observers are notified on a different thread.

Reply:
Good point, thanks